src/librustc_middle/mir/interpret/error.rs

```diff
@@ -454,6 +454,8 @@ pub enum UnsupportedOpInfo {
     ReadPointerAsBytes,
     /// Encountered raw bytes where we needed a pointer.
     ReadBytesAsPointer,
+    /// Accessing thread local statics
+    ThreadLocalStatic(DefId),
```

Comment:
This is not generally unsupported by the engine, just some `Machine` instances do not support it -- so maybe it should not be added here? `ConstEvalErrKind` might be a better place.

Reply:
The documentation on this enum states that some of the variants may be unused by miri and only be used by CTFE, so I assumed this was the right place.

Reply:
Ah... that comment hasn't been updated in a bit; this is certainly not the "fall-back error kind" any more.

However, you are right that `ReadPointerAsBytes`/`ReadBytesAsPointer` might also be impossible to emit from Miri at some point (currently it is still possible). So adding it here seems fine. (And maybe I should un-do some of my efforts to move error variants elsewhere? That would be https://github.com/rust-lang/rust/pull/70241 and to some extend https://github.com/rust-lang/rust/pull/70267.)

Please add a separator with comments here to separate "things the core engine does not support" from "things compile-time eval does not support". Miri-the-tool should ICE if it ever encounters one of the latter.